src/components/scene/background.js

```diff
@@ -1,49 +1,212 @@
+      this.probeLight = probeLight;
+    }
+  },
+  startLightProbe: function () {
+    var data = this.data;
+    var scene = this.el.sceneEl.object3D;
+
+    if (data.generateEnvironment) {
+      this.needsLightProbeUpdate = true;
+      this.needsEnvironmentUpdate = true;
+    } else {
+      scene.environment = null;
+    }
+  },
+  setupLightProbe: function () {
+    var scene = this.el.object3D;
+    var renderer = this.el.renderer;
+    var xrSession = renderer.xr.getSession();
+    var self = this;
+    var gl = renderer.getContext();
+
+    this.setupLightsForLightingEstimation();
+
+    this.glBinding = new XRWebGLBinding(xrSession, gl);
+    gl.getExtension('EXT_sRGB');
+    gl.getExtension('OES_texture_half_float');
+
+    xrSession.requestLightProbe()
+      .then(function (lightProbe) {
+        scene.environment = self.lightProbeTarget.texture;
+
+        self.xrLightProbe = lightProbe;
+        lightProbe.addEventListener('reflectionchange', function onReflectionChanged () {
+          self.needsEnvironmentUpdate = true;
+        });
+      })
+      .catch(function (err) {
+        console.warn('Lighting estimation not supported: ' + err.message);
+        console.warn('Are you missing: webxr="optionalFeatures: light-estimation;" from <a-scene>?');
+      });
   },
   update: function () {
     var scene = this.el.sceneEl.object3D;
     var data = this.data;
     var object3D = this.el.object3D;
+
     if (data.transparent) {
       object3D.background = null;
     } else {
       object3D.background = new THREE.Color(data.color);
     }
 
-    if (scene.environment && scene.environment !== this.cubeRenderTarget.texture) {
-      console.warn('Background will not overide predefined environment maps');
+    if (scene.environment &&
+      (scene.environment !== this.cubeRenderTarget.texture || scene.environment !== this.lightProbeTarget.texture)
+    ) {
+      console.warn('Background will not override user defined environment maps');
       return;
     }
 
     if (data.generateEnvironment) {
       scene.environment = this.cubeRenderTarget.texture;
+      this.needsEnvironmentUpdate = true;
     } else {
       scene.environment = null;
     }
   },
 
-  tick: function () {
-    if (!this.needsEnvironmentUpdate) return;
+  tick: function (time, delta) {
     var scene = this.el.object3D;
     var renderer = this.el.renderer;
-    var camera = this.el.camera;
+    var frame = this.el.sceneEl.frame;
+
+    this.timeSinceUpdate += delta;
+    if (
+      this.data.environmentUpdateFrequency > 0 && // should update in general?
+      this.timeSinceUpdate > (this.data.environmentUpdateFrequency * 1000) // should update this tick?
+    ) {
+      this.timeSinceUpdate = 0;
+      this.needsEnvironmentUpdate = true;
+    }
+
+    // Update Light Probe
+    // source: view-source:https://storage.googleapis.com/chromium-webxr-test/r886480/proposals/lighting-estimation.html
+    if (frame && this.xrLightProbe) {
```

Comment:
We can maybe factor out this part to an `updateLightProbe` method?

Reply:
I think this is still missing in the PR

Reply:
I took out the light updating logic and moved it to a function at the top I left the checks in place.